src/main/java/de/tum/in/www1/artemis/service/programming/ProgrammingExerciseGradingService.java

```diff
@@ -1090,7 +1090,7 @@ private void addFeedbackToStatistics(final Map<String, Integer> categoryIssuesMa
             final Feedback feedback) {
         if (feedback.isStaticCodeAnalysisFeedback()) {
             String categoryName = feedback.getText().substring(Feedback.STATIC_CODE_ANALYSIS_FEEDBACK_IDENTIFIER.length());
-            if ("".equals(categoryName)) {
+            if (categoryName.isEmpty()) {
```

Comment:
Are you sure that categoryName will never be null?

Reply:
Yes. You can see above that is's the result of `feedback.getText().substring()`. Since `feedback` cannot be null  it's never null here. 

I acutally left a few instances open where a call on isEmpty() would be no longer null-save, for these canges I'm positiv that null should never be a problem.